Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -964,26 +965,25 @@ - (void)cancelReadTimer
 - (void)viewDidAppear:(BOOL)animated
 {
     [super viewDidAppear:animated];
-    [self dismissKeyBoard];
-    [self startReadTimer];
 
+    [ProfileFetcherJob runWithThread:self.thread networkManager:self.networkManager];
+    [self markVisibleMessagesAsRead];
+    [self startReadTimer];
+    [self updateNavigationBarSubtitleLabel];
     [self updateBackButtonUnreadCount];
 
-    [self.inputToolbar endEditing:YES];
+    if (!self.viewHasEverAppeared) {
+        [self.inputToolbar endEditing:YES];
 
-    if (_composeOnOpen && !self.inputToolbar.hidden) {
-        [self popKeyBoard];
-        _composeOnOpen = NO;
-    }
-    if (_callOnOpen) {
-        [self callAction];
-        _callOnOpen = NO;
+        if (_composeOnOpen && !self.inputToolbar.hidden) {
+            [self popKeyBoard];
+            _composeOnOpen = NO;
+        }
+        if (_callOnOpen) {
+            [self callAction];
+            _callOnOpen = NO;
```

Comment:
I'm not sure we have to clear these flags since they are only applied if `! viewHasEverAppeared`

Reply:
Good point. I'm actually rethinking this... conceivably there's a world where we'd want to call/compose on appear without reloading the view. e.g. making a call from contacts app. I'm going to move the compose/call logic outside of this block.